serix/customtypes/serializable_set.go

```diff
@@ -0,0 +1,65 @@
+package customtypes
```

Comment:
I would leave it as is for now. The Set is an interface actually, and I don't know which concrete type should implement Encode()/Decode() methods because there are multiple Set implementations in hive.go.